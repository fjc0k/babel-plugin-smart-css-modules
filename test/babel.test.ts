import * as babel from '@babel/core'
import babelPluginSmartCssModules, { useSmartCssModules } from '../src'
import { join } from 'path'

describe('babel', () => {
  const fixtureFile = join(__dirname, './__fixtures__/index.js')

  test('[useSmartCssModules] default', async () => {
    const res = await babel.transformFileAsync(fixtureFile, {
      plugins: [useSmartCssModules()],
    })
    expect(res?.code).toMatchSnapshot()
  })

  test('[babelPluginSmartCssModules] default', async () => {
    const res = await babel.transformFileAsync(fixtureFile, {
      plugins: [babelPluginSmartCssModules],
    })
    expect(res?.code).toMatchSnapshot()
  })

  test('[useSmartCssModules] custom exts', async () => {
    const res = await babel.transformFileAsync(fixtureFile, {
      plugins: [
        useSmartCssModules({
          exts: ['.less'],
        }),
      ],
    })
    expect(res?.code).toMatchSnapshot()
  })

  test('[babelPluginSmartCssModules] custom exts', async () => {
    const res = await babel.transformFileAsync(fixtureFile, {
      plugins: [
        [
          babelPluginSmartCssModules,
          {
            exts: ['.less'],
          },
        ],
      ],
    })
    expect(res?.code).toMatchSnapshot()
  })

  test('custom flag', async () => {
    const res = await babel.transformFileAsync(fixtureFile, {
      plugins: [
        useSmartCssModules({
          flag: 'MMM',
        }),
      ],
    })
    expect(res?.code).toMatchSnapshot()
  })

  test('set ignore', async () => {
    const res = await babel.transformFileAsync(fixtureFile, {
      plugins: [
        useSmartCssModules({
          ignore: /2/,
        }),
      ],
    })
    expect(res?.code).toMatchSnapshot()
  })
})
