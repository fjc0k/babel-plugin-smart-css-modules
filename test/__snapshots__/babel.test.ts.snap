// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`babel [babelPluginSmartCssModules] custom exts 1`] = `
"import css from './0.css';
import less from \\"./2.less?modules\\";
import scss from './1.scss';
export default {
  css,
  less,
  scss
};"
`;

exports[`babel [babelPluginSmartCssModules] default 1`] = `
"import css from \\"./0.css?modules\\";
import less from \\"./2.less?modules\\";
import scss from \\"./1.scss?modules\\";
export default {
  css,
  less,
  scss
};"
`;

exports[`babel [useSmartCssModules] custom exts 1`] = `
"import css from './0.css';
import less from \\"./2.less?modules\\";
import scss from './1.scss';
export default {
  css,
  less,
  scss
};"
`;

exports[`babel [useSmartCssModules] default 1`] = `
"import css from \\"./0.css?modules\\";
import less from \\"./2.less?modules\\";
import scss from \\"./1.scss?modules\\";
export default {
  css,
  less,
  scss
};"
`;

exports[`babel custom flag 1`] = `
"import css from \\"./0.css?MMM\\";
import less from \\"./2.less?MMM\\";
import scss from \\"./1.scss?MMM\\";
export default {
  css,
  less,
  scss
};"
`;

exports[`babel set ignore 1`] = `
"import css from \\"./0.css?modules\\";
import less from './2.less';
import scss from \\"./1.scss?modules\\";
export default {
  css,
  less,
  scss
};"
`;
